http-clients/url-connection-client/src/main/java/software/amazon/awssdk/http/urlconnection/UrlConnectionHttpClient.java

```diff
@@ -320,7 +320,9 @@ private Optional<OutputStream> tryGetOutputStream() {
         }
 
         private Optional<InputStream> tryGetInputStream() {
-            return getAndHandle100Bug(() -> invokeSafely(connection::getInputStream), true);
+            return responseHasNoContent()
+                   ? Optional.empty()
+                   : getAndHandle100Bug(() -> invokeSafely(connection::getInputStream), true);
```

Comment:
Is it possible that we might need to check for no-content in tryGetErrorStream?

Reply:
It might be possible will do some deep dive , also will need to add a test case for this . Since I am going on leave will close this PR for now and will raise a separate PR when back from leave , will keep this task  is in fast track

Reply:
As per the code walk we donot need this check for erroStreams , details in https://github.com/aws/aws-sdk-java-v2/pull/3363